osis-core/src/main/java/com/scality/osis/service/impl/ScalityOsisServiceImpl.java

```diff
@@ -1042,7 +1040,46 @@ public ScalityOsisCaps updateOsisCaps(ScalityOsisCaps osisCaps) {
 
     @Override
     public PageOfOsisBucketMeta getBucketList(String tenantId, long offset, long limit) {
-        throw new NotImplementedException();
+        try {
+            logger.info("Get bucket list request received:: tenant ID:{}, offset:{}, limit:{}", tenantId, offset, limit);
+            Credentials tempCredentials = getCredentials(tenantId);
+            final AmazonS3 s3Client = this.s3.getS3Client(tempCredentials,
+                    appEnv.getRegionInfo().get(0));
+
+            //s3 listBucket has no pagination, so list all
+            List<Bucket> buckets = s3Client.listBuckets();
+
+            logger.debug("[S3] List all Buckets response:{}", new Gson().toJson(buckets));
```

Comment:
Just in case the returned array is large, I am not sure this line is doing any lazy evalutation: `toJson` could be a bit expensive to evaluate everytime, even if the log is not printed? (same for other `.debug` calls)

Reply:
good catch! I tried to do lazy logging using slf4j, but unfortunately, I didn't find a way.
so I just changed it to only log the bucket size, as the full content of all the buckets is that interesting to log